airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactory.kt

```diff
@@ -369,8 +379,13 @@ class ObjectStoragePathFactory(
                 pathVariableToPattern,
                 variableToIndex
             )
-        val combined = Path.of("$prefix/$replacedForPath").resolve(replacedForFile).toString()
-
+        // NOTE the old code does not actually resolve the path + filename,
+        // even tho the documentation says it does.
+        val combined = if (replacedForPath.startsWith('/')) {
+            "${prefix}$replacedForPath$replacedForFile"
+        } else {
+            "$prefix/$replacedForPath$replacedForFile"
+        }
```

Comment:
Is this different than `Paths.get`? I guess we are preserving terminal `/`?

Reply:
Yeah, `Paths.get()` will put a `/` between all its arguments and then conflate any adjacent `/`.

We want
* config path=`foo/` + config file=`bar` to be `foo/bar`
* BUT config path=`foo` + config file=`bar` to be `foobar`